apps/wallet/src/ui/app/helpers/SyncedStore.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+export class SyncedStore<T> {
```

Comment:
Why are we implementing our own store like this? Probably would be better to leverage something else instead of implementing this ourselves.

Reply:
I considered adding Jodai but the updates for the values are happening outside of react (might get a message from bg service even before react is initialised) so couldn't use it. Had a look at zunstand, but in the end decided to use this and decide on a library when we have more state to handle. I can change it to zunstand easily. Do you have in mind any alternatives?

Reply:
I used redux in the end, since it's already there

Reply:
Yeah let's just use redux for now. There are ways to make jotai work with state outside of react but it's a little bit of a chore. We can maybe take a holistic look at state management at some point soon.